adminSiteClient/VariablesAnnotationPage.tsx

```diff
@@ -1070,6 +1125,140 @@ class VariablesAnnotationComponent extends React.Component {
         }
         this.disposers = []
     }
+
+    @action.bound
+    updateFilterState(immutableTree: ImmutableTree, config: Config) {
+        const { filterState } = this
+
+        const updateFilterState = !(
+            isEqual(filterState?.tree, immutableTree) &&
+            isEqual(filterState?.config, config)
+        )
+
+        if (updateFilterState)
+            this.filterState = {
+                ...filterState,
+                tree: immutableTree,
+                config: config,
+            }
+    }
+
+    @computed get filterSExpression(): Operation | undefined {
+        const { filterState } = this
+        if (filterState) {
+            const tree = QbUtils.getTree(filterState.tree)
+            const sExpression = filterTreeToSExpression(tree)
+
+            return sExpression
+        }
+        return undefined
+    }
+
+    @computed get filterPanelConfigFields(): [string, SimpleField][] {
+        const { columnSelection, fieldDescriptions } = this
+        // The editors to use use are decided by two helper functions. For
+        // the more interesting fieldDescription mapping we rely on the
+        // preferred editor that is set in the editorOption field of the fieldDescription
+
+        if (fieldDescriptions === undefined) return []
+        const fieldDescriptionMap = new Map(
+            fieldDescriptions.map((fd) => [fd.pointer, fd])
+        )
+        const fields = excludeUndefined(
+            columnSelection.map((column) => {
+                if (isConfigColumn(column.key)) {
+                    return fieldDescriptionToFilterPanelFieldConfig(
+                        fieldDescriptionMap.get(column.key)!
+                    )
+                } else {
+                    return simpleColumnToFilterPanelFieldConfig(
+                        readOnlyColumnNamesFields.get(column.key)!
+                    )
+                }
+            })
+        )
+        return fields
+    }
+
+    @computed get FilterPanelConfig(): Config | undefined {
+        const { filterPanelConfigFields } = this
+
+        if (isEmpty(filterPanelConfigFields)) return undefined
+
+        const fieldsObject = Object.fromEntries(filterPanelConfigFields)
+        const config = {
+            ...filterPanelInitialConfig,
+            fields: fieldsObject,
+        }
+        // Hide operators in the UI that we don't have a good equivalent for
+        // in the S-Expressions
+        const operatorsToDrop = new Set([
+            "not_like",
+            "proximity",
+            "starts_with",
+            "ends_with",
+            "between",
+            "not_between",
+            "select_any_in",
+            "select_not_any_in",
+            "multiselect_equals",
+            "mutliselect_not_equals",
+        ])
+        config.types.text.widgets.field.operators =
```

Comment:
I have quickly tried to get rid of this godawful bunch of code, and the following seems to work:
`config.operators = omit(config.operators, operatorsToDrop) as any` (where omit comes from lodash)